From fb5effd09edac879701182dd5b730a471b173d37 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Wed, 21 Aug 2019 14:54:16 +0800
Subject: [PATCH 3/4] bcm2708_fb: fix compile warning

To avoid below warnings:

warning: format '%d' expects argument of type 'int', but argument 3 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]
drivers/video/fbdev/bcm2708_fb.c:536:22: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   unsigned char *p = (unsigned char *)ioparam.src + offset;
warning: format '%x' expects argument of type 'unsigned int', but argument 2 has type 'dma_addr_t' {aka 'long long unsigned int'} [-Wformat=]

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/video/fbdev/bcm2708_fb.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/fbdev/bcm2708_fb.c b/drivers/video/fbdev/bcm2708_fb.c
index 6c9ee9ca8ee2..1a1bfc98b0fe 100644
--- a/drivers/video/fbdev/bcm2708_fb.c
+++ b/drivers/video/fbdev/bcm2708_fb.c
@@ -524,7 +524,7 @@ static long vc_mem_copy(struct bcm2708_fb *fb, unsigned long arg)
 	buf = dma_alloc_coherent(fb->fb.device, PAGE_ALIGN(size), &bus_addr,
 				 GFP_ATOMIC);
 	if (!buf) {
-		pr_err("[%s]: failed to dma_alloc_coherent(%d)\n",
+		pr_err("[%s]: failed to dma_alloc_coherent(%zu)\n",
 				__func__, size);
 		rc = -ENOMEM;
 		goto out;
@@ -533,7 +533,7 @@ static long vc_mem_copy(struct bcm2708_fb *fb, unsigned long arg)
 	for (offset = 0; offset < ioparam.length; offset += size) {
 		size_t remaining = ioparam.length - offset;
 		size_t s = min(size, remaining);
-		unsigned char *p = (unsigned char *)ioparam.src + offset;
+		unsigned char *p = (unsigned char *)(unsigned long)ioparam.src + offset;
 		unsigned char *q = (unsigned char *)ioparam.dst + offset;
 
 		dma_memcpy(fb, bus_addr,
@@ -865,8 +865,8 @@ static int bcm2708_fb_probe(struct platform_device *dev)
 		goto free_fb;
 	}
 
-	pr_info("BCM2708FB: allocated DMA memory %08x\n",
-	       fb->cb_handle);
+	pr_info("BCM2708FB: allocated DMA memory %pad\n",
+	       &fb->cb_handle);
 
 	ret = bcm_dma_chan_alloc(BCM_DMA_FEATURE_BULK,
 				 &fb->dma_chan_base, &fb->dma_irq);
-- 
2.17.1

